### ISSUE TITLE
Freestyle libre + miaomiao2 - App crash when glucose level is high or low, can't add manual calibration

### ISSUE THREAD

**Issue by luk7227** [NightscoutFoundation/xDrip#1578]
Hi,

from couple days I have problem with xDrip when my glucose in xDrip is high or low (that's info from xDrip). 
The problem is that I have alarm from xDrip and when I try to open xDrip app is freezeing for couple seconds, so I use glucometer to add manual calibration to xDrip, but nothing changes and glucose level is still high in xDrip. Difference between glucometer and xDrip is 50 or 100 mg/dl, even when I use NFC and freestyle libre app there is difference like 50 or more, so sensor seems to work ok (I mean that when I check by Libre by NFC and glucometer levels are very close, but in xDrip is 50 or 100 more than in manual or NFC check).

**Tech details:**
**Phone**: Xiaomi Redmi 9 (MIUI global 11.0.9, Android 10 qp1A.190711.020), I have battery saving off, also allow xDrip to work in background, etc.
**Sensor**: Freestyle libre 1 (nfc)
**Add**: Miaomiao2
**XDrip version**: e14bae1-2020.07.13 (last stable)
**XDrip connection type:** Libre bluetooth

I have installed safety updates to Android couple days ago, but I didn't see that it changes anything to xDrip, I mean that there is no other issues than mentioned.

Also I cannot see any other issues, so when my glucose level is between low and high everything is working fine and glucose level is very close to glucometer check.

**Comment by luk7227** [comment]
One thing more, when glucose level is high or low in xDrip I need to restart phone or stop sensor (don't stop, just reset calibration) or it is still high or low, even if glucose level is ok ( f.e. after taling insulin and take some time).
If I try to not reset calibration in xDrip, xDrip still shows high or low with no changes. When I try to check with freestyle libre app by NFC sensor shows good level of sugar in blood (confirmed by glucometer check).

**Comment by luk7227** [comment]
Sesnor was broken, after couple days has stopped working. After use new one evertyhing seems to work fine.

**Comment by luk7227** [comment]
Hi,

unfortunately, I still have problem with disconnecting. Everything is still configured as above and I have new Libre 1 sensor with Miaomiao2 connected, but now XDrip is disconnecting Miaomiao2 and even if I can connect Miaomiao2 using bluetooth there is no data from sensor received.
I can only connect again when I going to stop sensor (don't stop, just reset calibration), otherwise no data is collected in XDrip and can't see sensor.

I have checked all logs and only thing that is supossed to repat before disconnecting is:

Calibration
Got invalid intercept value in xDrip classic algorithm: 56.06036681635072

(just after this is another)

Calibration
Null pointer in AdjustRecentReadings >=3: java.lang.NullPointerException: Attempt to read from field 'double.eveningoutpost.dexdrip.Model.Calibration.slope' on a null object reference.



![xdrip02 01 2021](https://user-images.githubusercontent.com/76660722/103467095-c05ae380-4d4b-11eb-8e1f-c24dc4bb8ca0.jpg)

**Comment by tzachi-dar** [comment]
Can you go to stop sensor.
Delete all calibrations and try a new calibration.

בתאריך שבת, 2 בינו׳ 2021, 23:43, מאת luk7227 ‏<notifications@github.com>:

**Comment by luk7227** [comment]
Hi, 

thanks for quick answer, I mentioned this above, when I go to stop sensor -> don't stop, just reset calibration, sensor is working again, but after some time sensor is going to stop working again and I have to do it again and again.

Sometimes it's 2 or 3 times a day, so I don't have information from sensor for couple hours or even all night, no alarms, nothing, so that's the biggest problem.

**Comment by tzachi-dar** [comment]
So, let me see if I got this correctly.
You have everything working (that is you are getting data) but then you
compare your bg to xDrip and there are big differences, so you calibrate,
and then things stop working?

I guess that in this case, using oop1 should be a good possibility.

See instructions her on how to enable it:
https://miaomiaoreader.medium.com/how-to-make-libre-us-14-days-sensors-work-with-miaomiao2-on-xdrip-35b431a40940

Thanks
Tzachi

**Comment by luk7227** [comment]
Sorry, I described two separate things, so first is already solved:
1. I had problem with low/high info and after this xDrip has stopped, so I need to reset phone, etc, but it seems that Libre sensor was broken. So this is closed, I changed Libre sensor and this problem disappear.

2. I still have have problem with:
I got new Libre sensor and have reset miaomiao2, but xDrip is disconnecting miaomiao2 and sensor and doesn't want to connect again. When I reset bluetooth or phone or miaomiao2 I can connect to miaomiao2, but xDrip can't find sensor.

Only thing that works is when I stop sensor and add new calibration, but it's temporary, because it happens every now and then and the only thing I see in the logs is in the post above.
So the problem is "losing" the sensor in xDrip and not collecting data until I notice that it is not working and I have to stop sensor and add new calibration sometimes 2 or 3 times a day on completly random time.
